cognite/pygen/_query/processing.py

```diff
@@ -355,35 +357,51 @@ def _unpack_edge(
     ) -> dict[dm.NodeId, list[dict[str, Any]]]:
         step_properties = set(step.selected_properties or []) or None
         unpacked_by_source: dict[dm.NodeId, list[dict[str, Any]]] = defaultdict(list)
+        is_leaf_step = len(connections) == 0
         for edge in step.edge_results:
-            start_node = dm.NodeId.load(edge.start_node.dump())  # type: ignore[arg-type]
-            end_node = dm.NodeId.load(edge.end_node.dump())  # type: ignore[arg-type]
+            start_node, end_node = dm.NodeId.load(edge.start_node.dump()), dm.NodeId.load(edge.end_node.dump())  # type: ignore[arg-type]
             if edge_expression.direction == "outwards":
-                source_node = start_node
-                target_node = end_node
+                source_node, target_node = start_node, end_node
             else:
-                source_node = end_node
-                target_node = start_node
+                source_node, target_node = end_node, start_node
             # step.view_id means that the edge has properties
+            dumped_edge: dict[str, Any]
             if self._edges == "include" or step.view_id:
-                dumped = self.flatten_dump(
+                dumped_edge = self.flatten_dump(
                     edge, step_properties, as_data_record=self._as_data_record, type_key=self._edge_type_key
                 )
-                for connection_property, node_targets_by_source in connections:
-                    if target_node in node_targets_by_source:
-                        dumped[connection_property] = node_targets_by_source[target_node]
-
-                unpacked_by_source[source_node].append(dumped)
+                self._append_connections(dumped_edge, connections, target_node, edge.as_id())
+                dumped = [dumped_edge]
             elif self._edges == "identifier":
-                dumped = edge.as_id().dump(include_instance_type=False)
-                unpacked_by_source[source_node].append(dumped)
+                dumped_edge = edge.as_id().dump(include_instance_type=False)
+                self._append_connections(dumped_edge, connections, target_node, edge.as_id())
```

Comment:
Found bugs for these two cases as well. However, they are never exposed to the user, as this is an internal class, so it is not part of the changelog.